lib/mesh.js

```diff
@@ -411,41 +467,66 @@ Mesh.prototype.initialize = Promise.promisify(function(config, callback){
           _this.getPackagedModules(config, callback);
         },
         function(callback) {
+
+          _this.startupProgress("got packaged modules", 70);
+
           if (!config.autoload || process.env.SKIP_AUTO_LOAD) {
             return callback();
           }
           _this.log.$$DEBUG('searched for autoload');
           _this.loadPackagedModules(config, null, callback);
         },
         function(callback) {
+
+          _this.startupProgress("got autoload modules", 75);
+
           if (!config.autoload || !process.env.SKIP_AUTO_LOAD) {
             _this.log.$$DEBUG('initialized autoload');
           }
           _this._loadComponentSuites(config, callback);
         },
         function(callback) {
+
+          _this.startupProgress("loaded component suites", 75);
```

Comment:
long as the values make sense - it looked like they might have been repeated accidentally.

Does the page only start showing progress when it changes? Maybe that's the underlying issue - could it not show the last known status when it loads?